packages/autorest.python/autorest/codegen/serializers/builder_serializer.py

```diff
@@ -665,6 +665,8 @@ def _serialize_body_parameter(self, builder: OperationType) -> List[str]:
                 f"_{body_kwarg_name} = self._serialize.body({body_param.client_name}, "
                 f"'{body_param.type.serialization_type}'{is_xml_cmd}{serialization_ctxt_cmd})"
             )
+        elif self.code_model.options["models_mode"] == "dpg":
+            create_body_call = f"_{body_kwarg_name} = dumps({body_param.client_name}, cls=AzureJSONEncoder)"
```

Comment:
we want to have the AzureJSONEncoder do the dumps on the body, you can see the code I've added in the `version_tolerant_models` PR. If we dump before we get to the `AzureJSONEncoder`, we lose the `readonly` information. We then pass in our vendored `AzureJSONEncoder` as the `cls`

Reply:
Hi @iscai-msft , I copied your changes at https://github.com/Azure/azure-sdk-for-python/commit/ca5e2e91a839c9eb272e7edcd71888bff24dfc19#diff-1b293f1507150fce1b8cdb3de315595de03ed03e3f629793874c0505ca4fca51R150 to the model_base.py.jinja2 in my PR to work with `readonly`.

And:
because the current released azure-core don't provide a parameter to customize `cls` in `set_json_body` (https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/core/azure-core/azure/core/rest/_helpers.py#L154), I don't find a way to pass vendored `AzureJSONEncoder` down to azure.core.rest.HttpRequest.
That's the reason I dump json to content before passing it to HttpRequest. Do we have other options to use the vendored `AzureJSONEncoder`?